### PR TITLE
swtpc09: restore to working status, and add comments to help

### PR DESCRIPTION
**Pull request by 68bit**
Retested all of these and they are still working. Did not have a UniFLEX DMAF2
disk to test that one combination, but it should be close to working.

Added comments to help people make a start with these.

### CODE REVIEW

File: src/mame/swtpc/swtpc09.cpp [hunk 1]
```diff
@@ -68,23 +68,90 @@
...
+MAME emulates the floppy disk controller more thoroughly than some other
+common emulators for these systems. Other common emulators can ignore the disk
+density and disk side driven by the software. FLEX numbers sectors
+contiguously from one side to the other so a simple emulator need only map the
+track and sector numbers. Real disk images often mix single and double density
+formats, e.g. single density for the first track, and the software may not
+have been updated to correctly drive the side and density outputs, so disks
+for these other emulators might not not work in this MAME emulator. These disk
+images and their softare need to be updated - this is not working around
+limitations of this MAME emulator, these would not be expected to work on real
```
Comment (cuavas) [inline src/mame/swtpc/swtpc09.cpp:82]: “MAME” is not an adjective, and shouldn’t be genericised.  Please don’t use it that way as in “MAME emulator”.  MAME is a proper noun.

File: src/mame/swtpc/swtpc09.cpp [hunk 2]
```diff
@@ -68,23 +68,90 @@
...
+hardware either.
+
+However the disk controllers used by this emulator have some options that can
+help with this transition, help get these disk images running as-is. The disk
+density selection can be overwitten. The disk side selection can be overridden
```
Comment (cuavas) [inline src/mame/swtpc/swtpc09.cpp:87]: Do you mean “density selection can be overridden” or do you actually mean something in the image can be overwritten?

File: src/mame/swtpc/swtpc09.cpp [hunk 3]
```diff
@@ -68,23 +68,90 @@
...
-mess swtpc09d3 -window -flop1 UOS_3S_BOOT.dsk
-type D to boot uniflex. This one has potential to get harddisk support for uniflex in the future.
-
-mess swtpc09i  not working
+Options (untested, but the DMAF2 is tested with FLEX):
+ DMAF2/3 FDC clock: 2.0MHz
+ DMAF2/3 expected density: double density
+Type D to boot UniFLEX.
+
+mess swtpc09d3 -hard1 wd0.chd -hard2 wd1.chd -fdc:0 8dsdd -flop1 UOS_3S_BOOT.mfi
```
Comment (cuavas) [inline src/mame/swtpc/swtpc09.cpp:116]: MESS is not a thing any more.  Please don’t pretend it is.

File: src/mame/swtpc/swtpc09.cpp [hunk 4]
```diff
@@ -68,23 +68,90 @@
...
+mess swtpc09i -io1:dc5:fdc:0 35hd -flop1 FLEX292F.dsk
+Options:
+ DC5 Control register bit 7: Erroneous side select
+ DC5 FDC clock: 2.0MHz
+ DC5 Expected density: double density
+ DC5 FLEX expected sectors per side: 36
+ CPU clock: 4MHz
+Type U to boot into FLEX
+
+mess swtpc09i -hard /tmp/2048X18X32.chd
```
Comment (cuavas) [inline src/mame/swtpc/swtpc09.cpp:135]: Same here – MESS as a product is well and truly dead.

File: src/mame/swtpc/swtpc09.cpp [hunk 5]
```diff
@@ -68,23 +68,90 @@
...
+trailing info, past 2048X18X32X256. Then fill every second byte with zero as
+only the lower 8 bits of every 16 bit word are used. Then convert to a MAME
+CHD disk file, adding the geometry:
+ chdman createhd -o 2048X18X32.chd -i 2048X18X32.dsk -chs 2048,18,32 -s 512
+This software appears to cold start FLEX at 0xcd00 as opposed to 0xc800, and the
+PIA IDE ROM needs patching for that, and there is a machine option to do so.
+At the SBUG prompt enter Ctrl-P and 'E800', then 'G' and it should boot into FLEX.
+This software might not match the ROM and hardware here.
+
+mess swtpc09o -io1:dc5:fdc:0 qd -flop1 OS9BTBK3.DSK
```
Comment (cuavas) [inline src/mame/swtpc/swtpc09.cpp:151]: please change this as well.

**Review by cuavas — CHANGES_REQUESTED** [review]

File: src/mame/swtpc/swtpc09.cpp (hunk 2 above)
Comment (68bit) [reply src/mame/swtpc/swtpc09.cpp:87]: The option overrides the drive density selection line, it does not modify the image format. This option it not an impractical hardware modification as a pure double density format disk could be created and the drive density wired for double density, without the boot ROM or OS disk drive on the disk having been updated (might be programmed for single density track 0 etc).

File: src/mame/swtpc/swtpc09.cpp (hunk 1 above)
Comment (68bit) [reply src/mame/swtpc/swtpc09.cpp]: Yes, done.

File: src/mame/swtpc/swtpc09.cpp (hunk 2 above)
Comment (cuavas) [reply src/mame/swtpc/swtpc09.cpp:86]: Then correct the comment – don’t use “overwritten” if you mean “overridden”.

File: src/mame/swtpc/swtpc09.cpp [hunk 6]
```diff
@@ -68,23 +68,89 @@
...
+The MAME emulates the floppy disk controller more thoroughly than some other
+common emulators for these systems. Other common emulators can ignore the disk
+density and disk side driven by the software. FLEX numbers sectors
+contiguously from one side to the other so a simple emulator need only map the
+track and sector numbers. Real disk images often mix single and double density
+formats, e.g. single density for the first track, and the software may not
+have been updated to correctly drive the side and density outputs, so disks
+for these other emulators might not work in the MAME. These disk images
+and their softare need to be updated - this is not working around limitations
+of the MAME, these would not be expected to work on real hardware either.
```
Comment (cuavas) [inline src/mame/swtpc/swtpc09.cpp:82]: You’re still using MAME as though it’s a generic term – “might not work in the MAME”, “limitations of the MAME”.

Comment (68bit) [reply src/mame/swtpc/swtpc09.cpp]: The use seems typical of uses in the docs/ but happy to just not mention 'MAME' here.

Comment (MooglyGuy) [reply src/mame/swtpc/swtpc09.cpp]: No it's not.

File: src/mame/swtpc/swtpc09.cpp (hunk 2 above)
Comment (68bit) [reply src/mame/swtpc/swtpc09.cpp]: Done.

File: src/mame/swtpc/swtpc09.cpp (hunk 5 above)
Comment (68bit) [reply src/mame/swtpc/swtpc09.cpp]: Removed all mention of 'mess'.

File: src/mame/swtpc/swtpc09.cpp (hunk 6 above)
Comment (68bit) [reply src/mame/swtpc/swtpc09.cpp]: @MooglyGuy I had already avoided the use of 'MAME', so that was all unnecessary.

**Pull request merged**